Code/Framework/AzFramework/AzFramework/Spawnable/Script/SpawnableScriptAssetRef.cpp

```diff
@@ -97,31 +104,7 @@ namespace AzFramework::Scripts
 
     void SpawnableScriptAssetRef::OnSpawnAssetChanged()
     {
-        // Disconnect from the bus beforehand in case the new asset is not valid
-        AZ::Data::AssetBus::Handler::BusDisconnect();
-
-        if (m_asset.GetId().IsValid())
-        {
-            AZStd::string spawnableAssetFile;
-            StringFunc::Path::GetFileName(m_asset.GetHint().c_str(), spawnableAssetFile);
-            StringFunc::Path::ReplaceExtension(spawnableAssetFile, Spawnable::DotFileExtension);
-            AZ::u32 spawnableAssetSubId = SpawnableAssetHandler::BuildSubId(spawnableAssetFile);
-
-            if (m_asset.GetId().m_subId != spawnableAssetSubId)
-            {
-                AZ::Data::AssetId spawnableAssetId = m_asset.GetId();
-                spawnableAssetId.m_subId = spawnableAssetSubId;
-
-                m_asset = AZ::Data::AssetManager::Instance().FindOrCreateAsset<Spawnable>(
-                    spawnableAssetId, AZ::Data::AssetLoadBehavior::Default);
-            }
-            else
-            {
-                m_asset.SetAutoLoadBehavior(AZ::Data::AssetLoadBehavior::Default);
-            }
-
-            AZ::Data::AssetBus::Handler::BusConnect(m_asset.GetId());
-        }
+        SetAsset(m_asset);
```

Comment:
Can you update the description of the PR explaining what the old code was doing and why it isn't needed anymore? It'll make it easier to follow the PR.

Reply:
Yes I have a task for SpawnableScriptMediator unit tests, which will include using SpawnableScriptAssetRef.